lib/github_status_reporter.rb

```diff
@@ -92,6 +98,20 @@ def get_repo_status(repo_name)
         )
       end
 
+      def update_production_tag(repo_name)
+        full_repo_name = orgify_repo_name(repo_name)
+        deploy_ref = 'tags/' + production_release_tag(full_repo_name)
+
+        head_commit_id = octokit_client.refs(full_repo_name, primary_ref(full_repo_name)).object.sha
+        commit_at_tag = octokit_client.refs(full_repo_name, deploy_ref).object.sha
+
+        if head_commit_id != commit_at_tag
+          octokit_client.update_ref(full_repo_name, deploy_ref, head_commit_id)
+        else
+          raise RefAlreadyDeployed, "HEAD and tag commit SHAs match, ref already deployed"
+        end
```

Comment:
Perhaps we should do some data transform here to return the tag instead of the `update_ref` result to simplify the call site of this method
e.g. instead of digging into the response from the `octokit_client.update_ref` method at the call site
`tag = output[:ref].split('/')[2]`

we could instead do that here
```suggestion
        raise RefAlreadyDeployed, 'HEAD and tag commit SHAs match, ref already deployed' if head_commit_id == commit_at_tag
        
        update_ref_response = octokit_client.update_ref(full_repo_name, deploy_ref, head_commit_id)
        # return the updated tag name
        update_ref_response[:ref].split('/')[2]`
```

Reply:
The output of the call contains stuff like the URLs for the tag and commit too. I'll make this change, but it seems to me that having more info available at the call site is more useful...I might have included those links in the response if it was more straightforward to build links and I wasn't also expecting a follow up message from an action (or jenkins). 

Good call on the one liner for raising the exception.